### DISCUSSION TITLE
Bezier Gradient Mask Generator with Falloff

### DISCUSSION THREAD

**Discussion by melieconiek** [opentoonz/opentoonz#6800] [Ideas]
OpenToonz could use a bezier shape gradient generator within an adjustable falloff. Most compositors have an effect like this. It would be very useful for creating feathered, keyframeable masks.

The example below is the Roto node from Natron.

![mask-gradient](https://user-images.githubusercontent.com/42298081/147369752-990c57fd-2fde-479d-a648-8014ec652d9e.gif)

**Comment by beeheemooth** [comment]
This is undoubtedly a very cool thing, but when I remember how difficult it is to work with effects in Preview mode, I imagine how difficult it will be to customize and animate it.

**Comment by melieconiek** [comment]
@beeheemooth That's a separate issue. It wouldn't be worse than any of the other gradient generators.

**Comment by RodneyBaker** [comment]
@konero closed his feature request but it has many simularities to this request so want to link that:  #4107

**Comment by adamearle** [comment]
Would be super handy to have a tool like this in opentoonz. Would be especially amazing with the OT Tracking tool and making sure their functionality is tip top and works smoothly together.

**Comment by adamearle** [comment]
some natron code https://github.com/NatronGitHub/Natron/search?q=roto+feather

**Comment by adamearle** [comment]
Interesting Blender Addon https://github.com/lukas-blecher/AutoMask

**Comment by artisteacher** [comment]
The corridor gradient gets us closer to this but I'm not a fan of the widget behavior - it seems fussier than is necessary.

**Comment by adamearle** [comment]
> ec

I see no sense in pursuing the development of that feature. to me it seems very clumsy and a bit old. I believe the suggestion should put the corridor effect into retirement.

**Comment by flurick** [comment]
The "Shade" vector brush (aka TBiColorStrokeStyle) is very close to this visually.
![OT_vector_shade](https://user-images.githubusercontent.com/3645797/200815282-5463b14a-0d4c-4d6d-9951-939215212991.png)

Making a tweaked version could be done but - comments are in Italian and some of the graphics api changed.
(quads are apparently deprecated in OpenGL)
